### ISSUE TITLE
Runtime path error: libg2o_types_slam3d.so

### ISSUE THREAD

**Issue by alessandroTorresani** [RainerKuemmerle/g2o#480]
Hello,

I am having issues to link correctly the library libg2o_types_slam3d.so to a simple example. 

I successfully compiled G2O with the default cmake params
`cmake .. && make && make install`

Then, I linked the g2o libraries to my executable in the_CmakeList.txt_ file:
![image](https://user-images.githubusercontent.com/3661566/104615496-97124f80-5689-11eb-8154-465431bf5186.png)

However, this simple code does not work:
![image](https://user-images.githubusercontent.com/3661566/104615730-cd4fcf00-5689-11eb-928b-ad4da4ad857f.png)

The compilation terminates, but the executable cannot find the **libg2o_types_slam3d.so** library. 
`./main: error while loading shared libraries: libg2o_types_slam3d.so: cannot open shared object file: No such file or directory`

This is the output of the `ldd main`
![image](https://user-images.githubusercontent.com/3661566/104616098-36cfdd80-568a-11eb-86b9-c76969debaac.png)

However, the **libg2o_types_slam3d.so** is present in the g2o installation folder
`ls /usr/local/lib/`

![image](https://user-images.githubusercontent.com/3661566/104616614-ca091300-568a-11eb-8fb4-8eddf6d6e64f.png)

Am I doing something wrong when linking the library in the _CmakeList.txt_?

**Comment by RainerKuemmerle** [comment]
Can you read the so?
`readelf -Ws /usr/local/lib/libg2o_types_slam3d.so`

**Comment by vahagnIV** [comment]
I had a similar issue, try running
    sudo ldconfig

**Comment by alessandroTorresani** [comment]
@RainerKuemmerle Yes I suppose I can read it, this is the output (only the first lines):

```
readelf -Ws /usr/local/lib/libg2o_types_slam3d.so

Symbol table '.dynsym' contains 1115 entries:
   Num:    Value          Size Type    Bind   Vis      Ndx Name
     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND 
     1: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNSo3putEc@GLIBCXX_3.4 (2)
     2: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZN3g2o7Factory8instanceEv
     3: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND atan2@GLIBC_2.2.5 (3)
     4: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZN3g2o23HyperGraphElementActionclEPNS_10HyperGraph17HyperGraphElementEPNS0_10ParametersE
     5: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZNKSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE7compareEPKc@GLIBCXX_3.4.21 (4)
     6: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND _ZN3g2o16OptimizableGraph4Edge18setMeasurementDataEPKd
```

**Comment by alessandroTorresani** [comment]
@vahagnIV Thank you, `ldconfig` seems to fix the problem above. 

However, I got a segmentation fault when running this code
![image](https://user-images.githubusercontent.com/3661566/104934657-b02d4000-59aa-11eb-93ee-fb2b16841416.png)

The line `optimizer.addParameter(cam_params);` makes the program to crash. Is this still related to the problem I had?
